### ISSUE TITLE
NavMesh::update crash if Scene does not have Layer

### ISSUE THREAD

**Issue by NatWeiss** [cocos2d/cocos2d-x#13158]
Since cocos2d-x 3.7, if a `Scene` is created without a `Layer` as a child, the application will immediately crash with EXC_BAD_ACCESS in `NavMesh::update()`. The call stack is as follows:

```
Director::drawScene()
    Scene::stepPhysicsAndNavigation()
        NavMesh::update()
```

If `CC_USE_NAVMESH` is set to `0` then another crash will occur in `Physics3DWorld::stepSimulate()`.

**Comment by zilongshanren** [comment]
@super626 
please take a look at this issue, thanks.

**Comment by gstolarov** [comment]
might be related to https://github.com/cocos2d/cocos2d-x/issues/12906

**Comment by super626** [comment]
@NatWeiss 
On my MAC, I checkout cocos2d-x 3.7 and create a new project with it.
And modify HellowWorld::createScene() like this,
Scene\* HelloWorld::createScene()
{
    // 'scene' is an autorelease object
    auto scene = Scene::create();

```
// 'layer' is an autorelease object
```

//    auto layer = HelloWorld::create();

```
// add layer as a child to scene
```

//    scene->addChild(layer);

```
Size visibleSize = Director::getInstance()->getVisibleSize();
Vec2 origin = Director::getInstance()->getVisibleOrigin();
// add "HelloWorld" splash screen"
auto sprite = Sprite::create("HelloWorld.png");

// position the sprite on the center of the screen
sprite->setPosition(Vec2(visibleSize.width/2 + origin.x, visibleSize.height/2 + origin.y));

// add the sprite as a child to this layer
scene->addChild(sprite, 0);

// return the scene
return scene;
```

}
It does not crash.

Can you show me how can I reproduce this bug?
Thanks.

**Comment by NatWeiss** [comment]
Hey, not at a computer right now, but can you try it without autorelease? I remember doing it like this:

```
auto scene = new Scene;
scene->init();
// add other stuff if desired, but not Layers
Director::getInstance()->runWithScene(scene);
scene->release();
```

**Comment by gstolarov** [comment]
If it's related to https://github.com/cocos2d/cocos2d-x/issues/12906 then it's a Visual Studio Solution files and will not be relevant to XCode.

**Comment by super626** [comment]
Thanks @NatWeiss @gstolarov 
@gstolarov  is right. It relates to #12906.
It should be fixed on current v3 branch. You can add CC_ENABLE_BULLET_INTEGRATION=1 macros in your created project, or remove that macros from cocos2d project.

**Comment by super626** [comment]
@zilongshanren @pandamicro 
It is fixed by #12906 
It can be closed.
